### PR TITLE
For #12257 - Add LoginDelegate and CreditCardDelegate to replace PromptFeature parameters

### PR DESCRIPTION
**Pull request by Alexandru2909**
Replaced PromptFeature parameters for credit card and login pickers and their related callbacks with LoginDelegate and CreditCardDelegate.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/creditcard/CreditCardDelegate.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     * from the select credit card prompt
+     */
+    val onSelectCreditCard: () -> Unit
+}
+
+/**
+ * Default implementation for [CreditCardDelegate]
+ */
+class DefaultCreditCardDelegate(
+    override val creditCardPickerView: SelectablePromptView<CreditCardEntry>? = null,
```
Comment (jonalmeida) [inline components/feature/prompts/src/main/java/mozilla/components/feature/prompts/creditcard/CreditCardDelegate.kt:37]: Drive-by: Is there a reason you didn't want to add the defaults directly on the interface itself?

**Review by jonalmeida — COMMENTED** [review]
+1 for the initiative to clean this up! ⚡

**Review by Amejia481 — APPROVED** [review]
LGMT 🏅 !
I think we should add a entry to the changelog indicating that there is a breaking change and create a PR for Fenix to address the breakage :)
